.github/actions/migrate/action.yml

```
@@ -0,0 +1,17 @@
+name: Migrate Nx Workspace
+description: Migrate Nx Workspace to the specified version
+inputs:
+  nx-version:
+    description: Nx version to try to migrate
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Migrate
+      shell: bash
+      run: |
+        echo '{"migrations":[]}' > migrations.json
+        npx nx migrate {nx-version}
+        npm install --force --ignore-scripts
+        npx nx migrate --run-migrations
```

You can tweak the migrate command to remove the need of creating an empty migrations file. Additionally, it is frequent that migrate commands update linting rules. Running linting with fix usually solves many of those issues, and I would consider it as a needed step in an automated workflow:

```suggestion
        npx nx migrate --run-migrations --if-exists
        npx nx run-many --target=lint --all --fix
```